src/signals/incident/components/IncidentClassification/index.js

```diff
@@ -13,20 +15,23 @@ const IncidentClassification = () => {
   const dispatch = useDispatch();
 
   useEffect(() => {
-    get(`${configuration.CATEGORIES_ENDPOINT}${category}/sub_categories/${subcategory}`);
-  }, [category, subcategory, get]);
+    if (isAuthenticated()) {
+      history.push('/');
+    }
+    else {
+      get(`${configuration.CATEGORIES_ENDPOINT}${category}/sub_categories/${subcategory}`);
+    }
+  }, [category, subcategory, get, history]);
 
   useEffect(() => {
-    if (!data && !error) return;
-
-    if (data) dispatch(setClassification({ category, subcategory }));
-    history.push('/');
+    if (data && data.is_active) dispatch(setClassification({ category, subcategory }));
+    if (data || error) history.push('/');
```

Comment:
I would think that, since the combination of `category` and `subcategory` is not a valid one at this point in the code, it would be best to replace the history entry instead of pushing another one onto the stack. There is no point in going back one page anyway.

Reply:
agree, I used replace on history here now.